ethportal-peertest/src/scenarios/find.rs

```diff
@@ -143,14 +139,13 @@ pub async fn test_trace_recursive_find_content_for_absent_content(peertest: &Pee
     let client = &peertest.nodes[0].ipc_client;
     let (content_key, _) = fixture_header_with_proof();
 
-    let result = HistoryNetworkApiClient::trace_recursive_find_content(client, content_key)
+    let error = HistoryNetworkApiClient::trace_recursive_find_content(client, content_key)
         .await
-        .unwrap();
-
-    assert!(!result.utp_transfer);
-    assert_eq!(result.content, PossibleHistoryContentValue::ContentAbsent);
-    // Check that at least one route was involved.
-    assert!(result.trace.responses.len() > 1);
+        .unwrap_err()
+        .to_string();
+    assert!(error.contains("Content not found"));
```

Comment:
I spent a long time trying to figure out a way to access the underlying `ErrorObject` here, so we don't have to test via `string.contains("..")`. I wasn't able to figure it out, but if anybody has any ideas then this test can be improved significantly

Reply:
If I'm not wrong, the error type is `jsonrpsee_types::ErrorObjectOwned`, which means you can access code, which should be -32001.
I think checking for both code and `string.contains` is good enough.

Maybe I'm missing something.

Reply:
Yeah, to be honest, I think we're both missing something. My best guess is that there is some strange type-casting going on inside the automatic `rpc` scaffolding, but I haven't been able to figure it out. I also agree that it should be `ErrorObjectOwned` but all I see is `jsonrpsee_core::Error` which has no available methods ( except `to_string()`). I haven't been able to find a way to get at the underlying `ErrorObjectOwned`. But, I don't think it's worth any more effort since it only tidies up this code a bit. I added a check for `string.contains("-32001")` though, so all of our bases should be covered.